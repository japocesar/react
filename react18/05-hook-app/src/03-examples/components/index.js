export * from './LoadingQuote';
export * from './QuoteComponent';