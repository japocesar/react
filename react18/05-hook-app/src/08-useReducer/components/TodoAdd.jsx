import React, { useState } from 'react'
import { useForm } from '../../hooks';

export const TodoAdd = ({ onNewTodo }) => {

    // const [newTodo, setNewTodo] = useState('');
    const {  description, onInputChange, onResetForm } = useForm({
        description: ''
    });

    const onSubmit = (event) => {
        event.preventDefault();
        if ( description.trim().length < 3 ) return;

        const newTodo = {
            id: new Date().getTime(),
            done: false,
            description,
        };

        onNewTodo && onNewTodo( newTodo );
        onResetForm();
    };


  return (
    <form onSubmit={ onSubmit }>
        <input
            type="text"
            placeholder="What's to do?"
            className="form-control"
            name="description"
            value={ description }
            onChange={ onInputChange }
        />

        <button
            type="submit"
            className="btn btn-primary mt-3"    
        >
            Add
        </button>
    </form>
  )
}
