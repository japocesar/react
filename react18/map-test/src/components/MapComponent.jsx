import { GoogleMap, LoadScript, Marker } from '@react-google-maps/api';
import { useState } from 'react';

import { mapConfig, distanceCalculator } from '../helpers';

export const MapComponent = ({ cityName,  cityLatLng ,onPlaceMarker, gameOver }) => {

  const [isMarkerShown, setIsMarkerShown] = useState(false);
  const [isCityMarkerShown, setIsCityMarkerShown] = useState(false);
  const [markerPosition, setMarkerPosition] = useState({ lat: 52.50, lng: 13.2846501 });
  const [cityMarkerPosition, setCityMarkerPosition] = useState({ lat: 52.50, lng: 13.2846501 });
  const [distance, setDistance] = useState(0);

  const containerStyle = {
    width: '100%',
    height: '400px'
  };

  // const defaultProps = {
  //   center: {
  //     lat: 52.50,
  //     lng: 13.2846501
  //   },
  //   zoom: 14,
  // };

  const onMapClick = ({latLng}) => {
    setIsMarkerShown(true);
    setMarkerPosition(latLng);
    setCityMarkerPosition( cityLatLng );
  }

  const onPlace = () => {
    setIsCityMarkerShown(true);
    const fixMarkerPos = { lat: markerPosition.lat(), lng: markerPosition.lng() };
    const distancel = distanceCalculator( markerPosition.lat(), cityMarkerPosition.lat, markerPosition.lng(), cityMarkerPosition.lng );
    // console.log(`${distance} KM!`);
    setDistance(distancel);
  }

  const onNext = () => {
    setIsMarkerShown(false);
    setIsCityMarkerShown(false);
    setDistance(0);
    onPlaceMarker( distance );
  }

  return (
    <div className='container'>
        <h1>Found the City:</h1>
        <div className="alert alert-primary fs-1" role="alert">{ cityName }</div>

        <LoadScript
          googleMapsApiKey="AIzaSyBhqD5BcsPIUmky6_n7ZjoeiwB3mEGLqiY"
        >
          <GoogleMap
            center={{ lat: 52.50, lng: 13.2846501 }}
            mapContainerStyle={containerStyle}
            onLoad={map => {
              const bounds = new window.google.maps.LatLngBounds();
              map.fitBounds(bounds);
            }}
            options={{ styles: mapConfig, zoom: 5 }}
            onClick={ onMapClick }
          >
            <>
              { ( isCityMarkerShown ) ? <Marker position={ cityMarkerPosition } />: <></> }
              { ( isMarkerShown ) ? <Marker key='user' position={ markerPosition } /> : <></> }
            </>
          </GoogleMap>
        </LoadScript>

        <hr />
        {
          ( distance !== 0 && distance < 50 ) && (<div className="alert alert-primary" role="alert">Distance: { distance } KM</div>)
        }
        {
          ( distance !== 0 && distance > 50 ) && (<div className="alert alert-danger" role="alert">Distance: { distance } KM</div>)
        }

        { (isMarkerShown && !isCityMarkerShown && !gameOver) && (<button
          className='btn btn-primary'
          onClick={ onPlace }
        >
          Place
        </button>)}

        { (isCityMarkerShown && !gameOver) && (<button
          className='btn btn-primary'
          onClick={ onNext }
        >
          Next City
        </button>) }
    </div>
  )
}
