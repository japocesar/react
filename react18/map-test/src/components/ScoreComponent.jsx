import { ScoreCard } from "./score/ScoreCard"

export const ScoreComponent = ( { cityIndex, score, cityCounter, gameOver } ) => {
  return (
    <div className="container">
      <h1>Score</h1>

      <div className="row">
        <div className="col-12 col-lg-6">
          <ScoreCard title="Cities Placed" value={ cityIndex } />
        </div>

        <div className="col-12 mt-3 col-lg-6 mt-lg-0">
        {/* <div className="progress">
          <div className="progress-bar" role="progressbar" aria-label="Example with label" style={{width:`25%`}} aria-valuenow="25" aria-valuemin="0" aria-valuemax="100">25%</div>
        </div> */}
          <ScoreCard title="Kilometers Left" value={ score } />
        </div>
      </div>

      { gameOver && (
        <div className="row mt-5">
          <h2>GAME OVER</h2>
            <div className="col">
              <ScoreCard title="Your Score" value={ cityCounter } />
            </div>
        </div>
      )}
    </div>
  )
}
