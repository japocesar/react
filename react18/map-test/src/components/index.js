export * from './MapComponent';
export * from './ScoreComponent';