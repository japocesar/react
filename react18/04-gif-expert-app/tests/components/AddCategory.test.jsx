import { render, screen, fireEvent } from "@testing-library/react"
import { AddCategory } from "../../src/components/AddCategory"

describe('Tests in <AddCategory />', () => {
    test('should match with snapshot', () => {
        const { container } = render( <AddCategory onNewCategory={ () => {} } /> );
        expect( container ).toMatchSnapshot();
    })

    test('should change the input value', () => { 
        render( <AddCategory onNewCategory={ () => {} } /> );
        const input = screen.getByRole('textbox');
        
        fireEvent.input( input, { target: { value: 'Saitama' } } );
        expect( input.value ).toBe('Saitama');
        // screen.debug();
     })

     test('should call onNewCategory if input has value', () => { 
        const inputValue = 'Saitama';
        const onNewCategory = jest.fn();

        render( <AddCategory onNewCategory={ onNewCategory } /> );

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input( input, { target: { value: inputValue } } );
        fireEvent.submit( form );

        expect( input.value ).toBe(''); 

        expect( onNewCategory ).toHaveBeenCalled();
        expect( onNewCategory ).toHaveBeenCalledTimes(1);
        expect( onNewCategory ).toHaveBeenCalledWith( inputValue );
      })

      test('should not call onNewCategory if inputValue is empty', () => {
        const inputValue = 'a';
        const onNewCategory = jest.fn();

        render( <AddCategory onNewCategory={ onNewCategory } /> );
        
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input( input, { target: { value: inputValue } } );
        fireEvent.submit( form );

        expect( onNewCategory ).not.toHaveBeenCalled();

      })
})