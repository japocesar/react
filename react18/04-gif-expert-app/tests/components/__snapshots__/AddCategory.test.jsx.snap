// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AddCategory /> should match with snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
